core/src/main/java/hudson/model/RunParameterValue.java

```diff
@@ -88,9 +94,41 @@ public String getNumber() {
     	return r == null ? null : r[1];
     }
 
+    public class SerializableValue implements Serializable {
```

Comment:
It won't work as you expect, because the class is not static. It includes reference to the parent class, and hence it is not really serializable now